python/cudf/cudf/core/column/lists.py

```diff
@@ -360,14 +375,45 @@ def get(self, index: int) -> ParentType:
         1    5
         2    6
         dtype: int64
+
+        >>> s.list.get(2)
+        0    <NA>
+        1       5
+        2       6
+        dtype: int64
+
+        >>> s.list.get(2, default=0)
+        0   0
+        1   5
+        2   6
+        dtype: int64
+
+        >>> s.list.get([0, 1, 2])
+        0   1
+        1   4
+        2   6
+        dtype: int64
         """
-        min_col_list_len = self.len().min()
-        if -min_col_list_len <= index < min_col_list_len:
-            return self._return_or_inplace(
-                extract_element(self._column, index)
-            )
+        if is_scalar(index):
+            out = extract_element_scalar(self._column, cudf.Scalar(index))
         else:
-            raise IndexError("list index out of range")
+            index = as_column(index)
+            out = extract_element_column(self._column, as_column(index))
+
+        if not (default is None or default is cudf.NA):
+            # determine rows for which `index` is out-of-bounds
+            lengths = count_elements(self._column)
+            out_of_bounds_mask = (np.negative(index) > lengths) | (
```

Comment:
Can we use the unary negative operator `-index` instead of `np.negative`? My expectation was that a Column would have basic operations like that, just like Series. It seems like the comparator below `index >= lengths` works fine, so it surprises me that the unary negative isn't supported.

Reply:
Columns don't really have well-defined API boundaries. Historically we have invoked many of their internal methods from the Frame layer. Not all operators are currently supported, and some of my recent binops PRs have been working to improve that. Even `np.negative` probably only started working after #10509, which was just merged 2 weeks ago; prior to that I'm guessing that you would have had to do `-1 * index`. I think the action item here is to mark this as a TODO.

Reply:
I raised https://github.com/rapidsai/cudf/issues/10611.